WcaOnRails/app/javascript/edit-events/modals/TimeLimit.jsx

```diff
@@ -0,0 +1,277 @@
+import _ from 'lodash'
+import React from 'react'
+import ReactDOM from 'react-dom'
+import Modal from 'react-bootstrap/lib/Modal'
+import Radio from 'react-bootstrap/lib/Radio'
+
+import events from 'wca/events.js.erb'
+import formats from 'wca/formats.js.erb'
+import AttemptResultInput from './AttemptResultInput'
+import { centisecondsToString, roundIdToString } from './utils'
+import ButtonActivatedModal from 'edit-events/ButtonActivatedModal'
+
+class RadioGroup extends React.Component {
+  get value() {
+    let formGroupDom = ReactDOM.findDOMNode(this.formGroup);
+    return formGroupDom.querySelector('input:checked').value;
+  }
+
+  render() {
+    return (
+      <div ref={c => this.formGroup = c}>
+        {this.props.children.map(child => {
+          return React.cloneElement(child, {
+            name: this.props.name,
+            key: child.props.value,
+            checked: this.props.value == child.props.value,
+            onChange: this.props.onChange,
+          });
+        })}
+      </div>
+    );
+  }
+}
+
+function objectifyArray(arr) {
+  let obj = {};
+  arr.forEach(el => obj[el] = true);
+  return obj;
+}
+
+class SelectRoundsButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { selectedRoundsById: objectifyArray(props.selectedRoundIds) };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ selectedRoundsById: objectifyArray(nextProps.selectedRoundIds) });
+  }
+
+  reset = () => {
+    this.setState({ selectedRoundsById: objectifyArray(this.props.selectedRoundIds) });
+  }
+
+  getSelectedRoundIds() {
+    return Object.entries(this.state.selectedRoundsById).filter(([k, v]) => v).map(([k, v]) => k);
+  }
+
+  onOk = () => {
+    this.props.onChange();
+    this._modal.close({ skipUnsavedChangesCheck: true });
+  }
+
+  hasUnsavedChanges = () => {
+    return JSON.stringify(this.props.selectedRoundIds) != JSON.stringify(this.getSelectedRoundIds());
+  }
+
+  render() {
+    let { timeLimit, excludeRound, wcifEvents } = this.props;
+    let selectedRoundsById = this.state.selectedRoundsById;
+
+    let wcifRounds = [];
+    wcifEvents.forEach(otherWcifEvent => {
+      // Cross round cumulative time limits may not include other rounds of
+      // the same event.
+      // See https://github.com/thewca/wca-regulations/issues/457.
+      let excludeEventId = excludeRound.id.split("-")[0];
+      let otherEvent = events.byId[otherWcifEvent.id];
+      let canChangeTimeLimit = otherEvent.can_change_time_limit;
+      if(!canChangeTimeLimit || excludeEventId == otherWcifEvent.id) {
+        return;
+      }
+      wcifRounds = wcifRounds.concat(otherWcifEvent.rounds.filter(r => r != excludeRound));
+    });
```

Comment:
I implemented something inspired by this use of `flatMap`. Please take a look at it after I commit my changes.

> I don't see the reason for rounds.filter(r => r != excludeRound), because when otherWcifEvent is the excluded round's event (excludeEventId), then we don't continue.

Good point! I definitely wrote this in the other order (exclude round, and *then* exclude event, and I didn't get rid of the excludeRound stuff). I'll change `excludeRound` to `excludeEventId`, which should hopefully be a little cleaner.

Reply:
It's great =)